### ISSUE TITLE
Next JS images and nested dynamic routes returns 503 error ECONNREFUSED - Timeout

### ISSUE THREAD

**Issue by samgajjar** [firebase/firebase-tools#6943]
### [REQUIRED] Environment info
Next JS 13.4.7 as mentioned in documentation from firebase. Below is my package.json
```
  "dependencies": {
    "firebase": "^10.10.0",
    "next": "14.1.4",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "@emotion/cache": "^11.11.0",
    "@emotion/react": "^11.11.4",
    "@emotion/styled": "^11.11.0",
    "@mui/icons-material": "^5.15.14",
    "@mui/material": "^5.15.14",
    "@mui/material-nextjs": "^5.15.11",
    "axios": "^1.6.8",
    "classnames": "^2.5.1",
    "html-react-parser": "^5.1.9",
    "path": "^0.12.7",
    "react-icons": "^5.0.1",
    "sass": "^1.72.0",
    "sharp": "^0.33.3"
  }
```

**firebase-tools:** Installed the latest version based on the documentation - Firebase version 13.6.0

**Platform:** macOS Sonama 14.2.1

### [REQUIRED] Test case
Next JS project deployed on us-east1 which has cloud functions enabled. Details of the cloud functions are at the bottom

#### Issue 1 - Image not rendering with Next JS Image component
**`<img />`** tag does not render on my localhost with npm run dev but renders when I deploy to firebase and **`<Image />`** tag does not render when deployed to firebase - Infact when we try to access the image based on the url it times out and gives a 503 error 
```
<img src={Logo} alt="Vanguard Logo" height="60" width="60" />
{/* <Image src={Logo} alt="Vanguard Logo" style={{ height: '60px', width: '60px' }} /> */}
```
Image Timeout error on Chrome network tab
```
Request URL: https://vanguard-dev-741bf.web.app/_next/image?url=%2Fassets%2Fimages%2Fvanguard%2Fanalytics-2.jpg
Request Method: GET
Status Code: 503 Service Unavailable
Remote Address: 199.36.158.100:443
Referrer Policy: strict-origin-when-cross-origin
```
Error on Chrome console tab
```
GET https://vanguard-dev-741bf.web.app/_next/image?url=%2Fassets%2Fimages%2Fvanguard%2Fanalytics-2.jpg 503 (Service Unavailable)
content.js:123 [Violation] 'message' handler took 179ms
image:1 Unchecked runtime.lastError: The message port closed before a response was received.
```
Code from page.js
```
<div>
  <Link href="/">
    <img src={Logo} alt="Vanguard Logo" height="60" width="60" /> <!-- does not work on localhost but works on Firebase -->
    {/* <Image src={Logo} alt="Vanguard Logo" style={{ height: '60px', width: '60px' }} /> */} <!-- does not work on firebase but works on localhost -->
  </Link>
</div>
```

#### Issue 2 - Accessing nested route from NextJS also returns a 503 error message
Blog main page available at - https://vanguard-dev-741bf.web.app/blog (works fine)
When accessing the nested route within the blog at https://vanguard-dev-741bf.web.app/blog/artificial-intelligence/0 where the path following this pattern `/blog/<category>/<postId>` it times out like the Image tag as follows:

Code of the Blog listing where the link points to `/blog/<category>/<postId>`
```
blogPosts.map((post, index) => (
  <>
    {/* User Box for Grid item sizing */}
    <Grid item key={index} xs={12} md={6}>
      <Card key={index}>
        <Link href={`/blog/${post.categorySlug}/${postId}`} style={{ textDecoration: "none", color: "black" }}>
          <CardActionArea>
            <CardMedia
              sx={{ height: 250 }}
              image={`${post.url}`}
              title={post.title}
            />
            <CardContent>
              <Typography gutterBottom variant="h5" component="div">
                {post.title}
              </Typography>
            </CardContent>
          </CardActionArea>
        </Link>
        <CardContent>
          <Typography variant="body2" color="text.secondary">
            {post.subtitle}
          </Typography>
        </CardContent>
        <CardActions disableSpacing>
          <IconButton aria-label="add to favorites">
            <AiTwotoneLike />
          </IconButton>
          {post.likeCount}
        </CardActions>
      </Card>
    </Grid>
  </>
))
```
Code of Blog Post article access the postId from the params props
```
export default function BlogPost({ params }) {
  return (
    <>
      <BlogPostHero
        bgImage={blogPosts[params.postId].url}
        title={blogPosts[params.postId].title}
        category={blogPosts[params.postId].category}
        tags={blogPosts[params.postId].tags}
      />
      <Container>
        <Grid container spacing={4}>
          <Grid item xs={12} lg={8}>
            <div className="section">
              <h2 className="title">
                {parse(blogPosts[params.postId].subtitle)}
              </h2>
              {parse(blogPosts[params.postId].content)}
              <br />
              <Grid container>
                <Grid item xs={1}>
                  <img
                    alt="..."
                    src="/assets/images/vanguard/light-bulb.png"
                    width="50"
                    height="50"
                  />
                </Grid>
                <Grid item xs={11}>
                  {parse(blogPosts[params.postId].provokingQuestion)}
                </Grid>
              </Grid>
              <br />
              {parse(blogPosts[params.postId].endNote)}
            </div>
          </Grid>
          <Grid item xs={12} lg={4}>
            <RelatedPosts />
            <hr />
            <SubscriptionForm title={"Subscribe to our Blog"} />
          </Grid>
        </Grid>
      </Container>
      <div className="spacer " />
    </>
  );
}
```

Error on chrome console tab
```
Failed to load resource: the server responded with a status of 503 ()
content.js:123 [Violation] 'message' handler took 164ms
0-how-ai-transforms-prospect-calls-and-automates-lead-generation:1 Unchecked runtime.lastError: Could not establish connection. Receiving end does not exist.
0-how-ai-transforms-…s-lead-generation:1 Uncaught (in promise) Error: A listener indicated an asynchronous response by returning true, but the message channel closed before a response was received
background.js:76 {message: 'Could not establish connection. Receiving end does not exist.'}
background.js:76 {message: 'Could not establish connection. Receiving end does not exist.'}
```

### [REQUIRED] Steps to reproduce
Access firebase site - https://vanguard-dev-741bf.web.app

### [REQUIRED] Expected behavior
It should render the image with `<Image />` tag https://vanguard-dev-741bf.web.app/ (logo and images missing when scroll to content)
Render the Blog post when accessing https://vanguard-dev-741bf.web.app/blog/artificial-intelligence/0
 
Works perfectly fine on localhost

### [REQUIRED] Actual behavior
Any image with <Image /> should be rendered as it gets rendered on localhost:3000
Any page accessed from nested dynamic route should be rendered as it gets rendered on localhost:3000

Executed the debug with firebase serve --debug and here are the logs 
#### Issue 1 - Image not rendering
```
i  hosting: ::1 - - [01/Apr/2024:19:15:54 +0000] "GET /_next/image?url=%2Fassets%2Fimages%2Fvanguard%2Fartificial_intelligence.jpg&w=1080&q=75 HTTP/1.1" 500 - "http://localhost:5002/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/123.0.0.0 Safari/537.36" {"metadata":{"emulator":{"name":"hosting"},"message":"::1 - - [01/Apr/2024:19:15:54 +0000] \"GET /_next/image?url=%2Fassets%2Fimages%2Fvanguard%2Fartificial_intelligence.jpg&w=1080&q=75 HTTP/1.1\" 500 - \"http://localhost:5002/\" \"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/123.0.0.0 Safari/537.36\""}}
Error: connect ECONNREFUSED ::1:50346
    at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1495:16) {
  errno: -61,
  code: 'ECONNREFUSED',
  syscall: 'connect',
  address: '::1',
  port: 50346
}
```

#### Issue 2 - Cannot access nested routes page 
Localhost URL : http://localhost:5002/blog/artificial-intelligence/0
Firebase URL : https://vanguard-dev-741bf.web.app/blog/artificial-intelligence/0
```
Error: connect ECONNREFUSED ::1:50346
    at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1495:16) {
  errno: -61,
  code: 'ECONNREFUSED',
  syscall: 'connect',
  address: '::1',
  port: 50346
}
i  hosting: ::1 - - [01/Apr/2024:19:18:29 +0000] "GET /blog/artificial-intelligence/0 HTTP/1.1" 500 - "http://localhost:5002/blog" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/123.0.0.0 Safari/537.36" {"metadata":{"emulator":{"name":"hosting"},"message":"::1 - - [01/Apr/2024:19:18:29 +0000] \"GET /blog/artificial-intelligence/0 HTTP/1.1\" 500 - \"http://localhost:5002/blog\" \"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/123.0.0.0 Safari/537.36\""}}
```

Goole Cloud function created has the following URL and also return 504 error
https://ssrvanguarddev741bf-wk4my75kaa-ue.a.run.app/

Google Cloud function logs also just gives timeout
{
  "textPayload": "The request has been terminated because it has reached the maximum request timeout. To change this limit, see https://cloud.google.com/run/docs/configuring/request-timeout",
  "insertId": "660b11190002a277f2cf86f4",
  "httpRequest": {
    "requestMethod": "GET",
    "requestUrl": "https://fh-d9c4e07707d91bec---ssrvanguarddev741bf-wk4my75kaa-ue.a.run.app/blog/artificial-intelligence/0",
    "requestSize": "1757",
    "status": 504,
    "responseSize": "714",
    "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/123.0.0.0 Safari/537.36",
    "remoteIp": "74.125.212.3",
    "referer": "https://vanguard-dev-741bf.web.app/blog",
    "serverIp": "216.239.38.53",
    "latency": "59.995129514s",
    "protocol": "HTTP/1.1"
  },
  "resource": {
    "type": "cloud_run_revision",
    "labels": {
      "revision_name": "ssrvanguarddev741bf-00019-guv",
      "location": "us-east1",
      "service_name": "ssrvanguarddev741bf",
      "project_id": "vanguard-dev-741bf",
      "configuration_name": "ssrvanguarddev741bf"
    }
  },
  "timestamp": "2024-04-01T19:54:05.170868Z",
  "severity": "ERROR",
  "labels": {
    "instanceId": "00a22404dc1bf313e911763ed6a4587c11cdf345e1d818e0640235b7eb624042824cebea413b5c4f0f7c7683c37f2fbd120fd087316a3258837fd2a8cdd64a3f",
    "goog-managed-by": "cloudfunctions"
  },
  "logName": "projects/vanguard-dev-741bf/logs/run.googleapis.com%2Frequests",
  "trace": "projects/vanguard-dev-741bf/traces/b07fccff823cef75971c80c21f85ca3e",
  "receiveTimestamp": "2024-04-01T19:55:05.408700821Z",
  "spanId": "1793253968491586128"
}

**Comment by samgajjar** [comment]
I create a fresh blank next js app and added a very simple nested route and tested with firebase serve and it worked. So I copied one file at a time and kept on checking the firebase serve and all seems to work.
Finally when I copied all the pages and routes then thoroughly tested and then deployed to firebase and everything is working

Not sure what changed.. May be something was messed up in my previous project folder.
